src/test/java/org/ambraproject/wombat/service/PeerReviewServiceTest.java

```diff
@@ -0,0 +1,130 @@
+      @Override
+      String getReviewXml(Map<String, ?> metadata) throws IOException {
+        Map<String, ?> files = (Map<String, ?>) metadata.get("files");
+        Map<String, ?> letter = (Map<String, ?>) files.get("letter");
+        String crepoKey = (String) letter.get("crepoKey");
+
+        String letterContent = null;
+        if (crepoKey == "info:doi/10.1371/journal.pone.0207232.r001.xml") {
+          letterContent = "<?xml version=\"1.0\" encoding=\"utf-8\"?><sub-article specific-use=\"decision-letter\">" +
+              "<front-stub><custom-meta-group><custom-meta><meta-name>Submission Version</meta-name><meta-value>0</meta-value></custom-meta></custom-meta-group></front-stub>" +
+              "<body><p>InitialDecisionLetterSampleBody</p></body></sub-article>";
+        }
+        if (crepoKey == "info:doi/10.1371/journal.pone.0207232.r002.xml") {
+          letterContent = "<?xml version=\"1.0\" encoding=\"utf-8\"?><sub-article article-type=\"author-comment\">" +
+              "<front-stub><custom-meta-group><custom-meta><meta-name>Submission Version</meta-name><meta-value>1</meta-value></custom-meta></custom-meta-group></front-stub>" +
+              "<body><p>FirstRoundAuthorResponseSampleBody</p></body></sub-article>";
+        }
+        if (crepoKey == "info:doi/10.1371/journal.pone.0207232.r003.xml") {
+          letterContent = "<?xml version=\"1.0\" encoding=\"utf-8\"?><sub-article specific-use=\"decision-letter\">" +
+              "<front-stub><custom-meta-group><custom-meta><meta-name>Submission Version</meta-name><meta-value>1</meta-value></custom-meta></custom-meta-group></front-stub>" +
+              "<body><p>FirstRoundDecisionLetterSampleBody</p></body></sub-article>";
+        }
+        return letterContent;
+      }
+    };
+
+    String html = serviceWithMockedContent.asHtml(itemTable);
+    Document d = Jsoup.parse(html);
+
+    assertThat(d.select(".review-history th").get(0).text(), containsString("Original Submission"));
+    assertThat(d.select(".review-history th").get(1).text(), containsString("Revision 1"));
+    assertThat(d.select(".review-history .decision-letter").get(0).text(), containsString("InitialDecisionLetterSampleBody"));
+    assertThat(d.select(".review-history .decision-letter").get(1).text(), containsString("FirstRoundDecisionLetterSampleBody"));
+    assertThat(d.select(".review-history .author-response").get(0).text(), containsString("bogus"));
+  }
+
+  @Test
+  public void testAsHtmlHandlesNoPeerReviewItems() throws IOException {
+    ImmutableMap<String, ? extends Map<String, ?>> itemTable = ImmutableMap.of(
+        "10.1371/journal.pone.0207232.t001", ImmutableMap.of(
+            "doi", "10.1371/journal.pone.0207232.t001",
+            "itemType", "table"
+        )
+    );
+    String html = new PeerReviewServiceImpl().asHtml(itemTable);
+    assertNull(html);
+  }
+
+  @Test
+  public void testXslt() {
```

Comment:
How about we delete testXslt() from here. A more robust version will be added in AMBR-657 (testAuthorResponse()).

Reply:
done!